clients/src/main/java/org/apache/kafka/clients/consumer/internals/TopicMetadataRequestManager.java

```diff
@@ -55,16 +56,13 @@
  * <p>
  * The manager checks the state of the {@link TopicMetadataRequestState} before sending a new one to
  * prevent sending it without backing off from previous attempts.
- * It also checks the state of inflight requests to avoid overwhelming the broker with duplicate requests.
- * The {@code inflightRequests} are memorized by topic name. If all topics are requested, then we use {@code Optional
- * .empty()} as the key.
- * Once a request is completed successfully, its corresponding entry is removed.
+ * Once a request is completed successfully or times out, its corresponding entry is removed.
  * </p>
  */
 
 public class TopicMetadataRequestManager implements RequestManager {
     private final boolean allowAutoTopicCreation;
-    private final Map<Optional<String>, TopicMetadataRequestState> inflightRequests;
+    private final List<TopicMetadataRequestState> inflightRequests;
```

Comment:
Why did you decide to remove the reuse of requests

Reply:
Because it seemed to me like an unnecessary complexity that wasn't being used in practice. The `listTopics` and `partitionsFor` methods are necessarily used by the single application thread that is allowed to use the `KafkaConsumer`. The calls do not overlap.